spec/index.md

```diff
@@ -1550,6 +1557,16 @@ If this traps (e.g. runs out of cycles), the state changes from the `cleanup` fu
 
 There must be at most one call to `ic0.call_on_cleanup` between `ic0.call_new` and `ic0.call_perform`.
 
+-   `ic0.call_set_timeout : (seconds : i32) -> ()`
+
+    Requests the system to generate a synthetic timeout reject if no reply is received within the specified amount of seconds. The returned reject code will be `SYS_TIMEOUT`.
+
+    The `seconds` parameter is only advisory. First, it is silently bounded by the `MAX_CALL_TIMEOUT` system constant; i.e., larger timeouts are treated as equivalent to `MAX_CALL_TIMEOUT` and do not cause an error. Second, while the system will attempt not to time out the call too early, it may in exceptional circumstances trigger the timeout earlier than requested; the calling canister should not rely on the timeout not occurring early. Finally, even under normal circumstances, the timeout may trigger significantly later than the specified time.
```

Comment:
I'd treat the "in exceptional circumstances" as an implementation detail and not mention it here. Mainly to avoid the situation where somebody reads this as "it is fine if the canister can't handle this case as it will never happen". 
```suggestion
    The `seconds` parameter is only advisory. First, it is silently bounded by the `MAX_CALL_TIMEOUT` system constant; i.e., larger timeouts are treated as equivalent to `MAX_CALL_TIMEOUT` and do not cause an error. Second, it may (if the load on the subnet demands it) trigger the timeout earlier than requested; the calling canister should not rely on the timeout not occurring early. Finally, even under normal circumstances, the timeout may trigger significantly later than the specified time.
```

Reply:
Well, it explicitly says in that sentence that you should not rely on the timeout not triggering early :) But I can change, I just wouldn't mention subnets since we so far don't have a real treatment of them in the document.